### ISSUE TITLE
Resume error not fixed yet

### ISSUE THREAD

**Issue by iripu** [downthemall/downthemall#176]
### Your DownThemAll! and Browser Version and Language and Operating System
DTA 3.0.8, Firefox 48.0.2 32bit PortableApps, Windows 8.1 core 64bit, en-GB both FF and OS

### Expected Behavior
There are many websites that changes download links periodically. The files allow pausing and continue. But the download links stop working after that period. Suppose I am downloading a 1 GB file. Due to some reason I can't complete it and it was paused at 80%. I come back after a day and try to resume it. But, the download link has already stopped working. The file is now available at a different link. DTA will now show an error and all that 800mb will be deleted. I wish DTA to keep it at the paused state and may be show a message that the file is not available at that location now. And if I enter a new download link, it will continue from that 80% just like normal.

### Actual Behavior
After error, can't resume download. It deletes whatever amount has been downloaded and deletes the chunk bits information.

### Steps To Reproduce
Start downloading any Youtube video (I use Flashgot extension), pause at about 10%, close Firefox if you want. After about 24 hrs, try to resume that file. It will show error and you will not be able to continue even if you enter a new download link.

### Additional info such as log extracts
*Please enable Diagnostic Logging in the Privacy tab of the DownThemAll! preferences and either copy/paste essential part of the log or attach the entire log. Please note that the log may include sentitive information which you have to remove prior to posting it!*
[DTA_log.txt](https://github.com/downthemall/downthemall/files/696381/DTA_log.txt)

**Comment by iripu** [comment]
I know that workaround. But the issue seems simple to fix, just don't delete the file. DTA is around for so long, this issue should have been fixed years ago.

**Comment by Gabr-F** [comment]
Yeah, in any circumstance automatically deleting anything in case of errors seems stupid.
What if it took one weeks to get to that point, and he knew a way to complete the download if only what had been downloaded so far hadn't been deleted?

I presume this is an accidental behaviour, but it's quite serious and ought to be corrected. It seems there isn't a lot moving in this project lately though.

**Comment by nmaier** [comment]
Leaving errored out files around is not an option either. Other download tools, in particular the built-in downloaders of browser, remove the partial (temp) downloads too in case of a hard error such as 404.

**Comment by Gabr-F** [comment]
@nmaier What does that mean? If other downloaders are stupid this one has to be too?

**Comment by iripu** [comment]
@nmaier What's the point if it follows what built-in downloader does? Desktop download managers never delete files like DTA does.


I can just add a new destination and resume the downloading. The "errored out" file is neither corrupted nor it does act like a malware, that you must have to delete it to save the user's PC. It's just paused and can be resumed with a new link. It's not "hard error".


I didn't expect this kind of response.
